src/main/java/bio/terra/stairway/Stairway.java

```diff
@@ -119,17 +125,33 @@ public boolean quietDown(long waitTimeout, TimeUnit unit) {
         try {
             return threadPool.awaitTermination(waitTimeout, unit);
         } catch (InterruptedException ex) {
-            Thread.currentThread().interrupt();
+            Thread.interrupted();
             return false;
         }
     }
 
     /**
      * Not-so-graceful shutdown: shutdown the pool which will cause an InterruptedException on all of
      * the flights. That _should_ cause the flights to rapidly terminate and get set to unowned.
+     * @param waitTimeout time, in some units to wait before timing out
+     * @param unit the time unit of waitTimeout.
+     * @throws InterruptedException on interruption during termination
      */
-    public void terminate() {
-        threadPool.shutdownNow();
+    public boolean terminate(long waitTimeout, TimeUnit unit) throws InterruptedException {
+        List<Runnable> neverStartedFlights = threadPool.shutdownNow();
+        for (Runnable flightRunnable : neverStartedFlights) {
+            Flight flight = (Flight) flightRunnable;
+            try {
+                logger.info("Setting never-started flight ready: " + flight.context().getFlightId());
+                flight.context().setFlightStatus(FlightStatus.READY);
+                flightDao.exit(flight.context());
+            } catch (DatabaseOperationException | FlightException ex) {
+                // Not much to do on termination
+                logger.warn("Unable to exit never-started flight: " + flight);
```

Comment:
Could you include the flightid here, as well? For debugging purposes, just thinking about when I'm querying PostGres.